imagescan/delta.go

```diff
@@ -258,14 +264,14 @@ func (d *deltaState) getImages() []*image {
 }
 
 func (d *deltaState) updateImage(i *image, change func(img *image)) {
-	img := d.images[i.cacheKey()]
+	img := d.images[getImageKey(i.id, i.architecture, i.name)]
```

Comment:
image contains key field. In all these examples it can be simplified to use i.key
We set key only once during initial image write to map.

Same for all tests.

Maybe getImageKey can be renamed to newImageKey or even removed from and created directly inline in that single place there it's first time written.